source/implementations/f7/Meadow.F7/Devices/F7GPIOManager_interrupts.cs

```diff
@@ -188,37 +189,40 @@ private void InterruptServiceThreadProc(object o)
             if (_firstInterrupt)
             {
                 // DEV NOTE: this is to force the interp pipeline top build at least some of the call stack and improve the response of first-interrupt results
-                Interrupt?.Invoke(new NullPin(), false);
+                Interrupt?.Invoke(_nullPin, false);
                 _firstInterrupt = false;
             }
 
             int priority = 0;
-            try
+
+            var result = Interop.Nuttx.mq_receive(queue, rx_buffer, rx_buffer.Length, ref priority);
+
+            // byte 1 contains the port and pin, byte 2 contains the stable state.
+            if (result >= 0)
             {
-                var result = Interop.Nuttx.mq_receive(queue, rx_buffer, rx_buffer.Length, ref priority);
+                var irq = rx_buffer[0];
+                bool state = rx_buffer[1] != 0;
+                var port = irq >> 4;
+                var pin = irq & 0xf;
+                IPin ipin;
+
 
-                // byte 1 contains the port and pin, byte 2 contains the stable state.
-                if (result >= 0)
+                lock (_interruptPins)   //FIXME: If this is supposed to keep modifications from
+                                        //happening on the array, lock()'s are needed at those points also
                 {
-                    var irq = rx_buffer[0];
-                    bool state = rx_buffer[1] != 0;
-                    var port = irq >> 4;
-                    var pin = irq & 0xf;
-
-                    lock (_interruptPins)
-                    {
-                        var ipin = _interruptPins[port, pin];
-                        if (ipin != null)
-                        {
-                            Interrupt?.Invoke(ipin, state);
-                        }
-                    }
+                    ipin = _interruptPins[port, pin];
+                    if (ipin == null)
+                        continue;
+                }
+
+                try
+                {
+                    Interrupt?.Invoke(ipin, state);
+                }
+                catch (Exception ex)
+                {
+                    Thread.Sleep(5000);
```

Comment:
I feel like this isn't what we want, though offhand I'm not certain of a replacement.  the App's interrupt handler has had an unhandled exception. Should we LogError?  We can't throw, or our loop crashes, but this causes all interrupts to halt for 5 seconds which also feels like a bad experience.

Reply:
I don't understand the 5 second halt (to be clear, it was already there), but LogError sounds right, and useful for diagnostics.

Reply:
Ah yes, I did remove Resolver.Log.Error. Any way we can have a supersafe variant that doesn't do I/O?